src/github.com/getlantern/flashlight/client/config.go

```diff
@@ -25,6 +26,24 @@ func (c *ClientConfig) SortServers() {
 	sort.Sort(ByHost(c.FrontedServers))
 }
 
+// HighestQOSFrontedDialer returns the fronted.Dialer with the highest QOS.
+func (c *ClientConfig) HighestQOSFrontedDialer() fronted.Dialer {
+	var highestQOSFrontedDialer fronted.Dialer
+	highestQOS := math.MinInt32
+	for _, s := range c.FrontedServers {
+		// Get a dialer for domain fronting (fd) and a dialer to dial to arbitrary
+		// addreses (dialer).
+		fd, dialer := s.dialer(c.MasqueradeSets)
```

Comment:
A lot of `fronted.Dialer`s are created and discarded here.  Only the one that is returned has a chance of ever being `Close`d.  The code prior to this was storing the `hqfd` in the client so it could be closed.  Maybe these only need closing if they're ever used?